release notes/v0.31.0.md

```diff
@@ -0,0 +1,75 @@
+k6 v0.31.0 is here! :tada: It's a smaller release with some performance improvements, a new `http_req_failed` metric and changes to the output subsystem that enables output extensions with xk6!
+
+## New features
+
+### Output cleanup and registration API ([#1874](https://github.com/loadimpact/k6/pull/1874))
+
+The state of k6's output packages has been a development pain point for a long time which made it difficult to add new outputs in a consistent way. In the refactor done in v0.31.0 this has been mostly addressed and outputs now implement a saner `Output` interface.
+
+In addition to this, an output registry was added to allow external outputs to be easily added. This expands the possibilities of what [xk6](https://github.com/k6io/xk6) can do, so now besides allowing external JS modules, it's also possible to build custom k6 binaries that use external output implementations! This is very useful if you use a system that's currently not supported by [the built-in outputs](https://k6.io/docs/getting-started/results-output), or need some custom handling of the metrics k6 produces.
+
+We are working on the proper documentation for this, as well as the overdue xk6 documentation about JS extensions, so keep a lookout on https://k6.io/docs/ .
```

Comment:
I considered adding a small output extension example, but it involves a lot of explanation for how e.g. the `SampleBuffer` or `PeriodicFlusher` is used, so I think it's best if we leave that for the docs.

Reply:
A full example will be overkill here, but I added a change suggestion above with links to the new interface. I am pretty confident that for a reasonably skilled Go developer that will be enough to start with, and we can help with any questions on the forum, github or slack until we have the proper docs up...